ledger/src/blockstore.rs

```diff
@@ -136,10 +136,27 @@ impl std::fmt::Display for InsertDataShredError {
     }
 }
 
+/// A "complete data set" is a range of [`Shred`]s that combined in sequence carry a singled
+/// serialized [`Vec<Entry>`].
+///
+/// Services such as `solana_core::window_service::WindowService` for a TVU, and
+/// `solana_core::replay_service::ReplayStage` for a TPU, drive generation of these, via direct or
+/// indirect calls to [`Blockstore::insert_shreds_handle_duplicate()`].
+///
+/// `solana_core::completed_data_sets_service::CompletedDataSetsService` is the main receiver of
+/// `CompletedDataSetInfo`.
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub struct CompletedDataSetInfo {
+    /// [`Slot`] for which we have received the last [`Shred`] in the `start_index..=end_index`
```

Comment:
nit: `[`Slot`] for which we have received the last [`Shred`] in the `start_index..=end_index` -> [`Slot`] to which the shreds in this set belong